modules/jupyter/main.tf

```diff
@@ -63,17 +63,32 @@ module "iap_auth" {
 module "jupyterhub-workload-identity" {
   source              = "terraform-google-modules/kubernetes-engine/google//modules/workload-identity"
   use_existing_gcp_sa = !var.create_service_account
-  name                = var.gcp_service_account
+  name                = var.gcp_and_k8s_service_account
   namespace           = var.namespace
   project_id          = var.project_id
-  roles               = ["roles/cloudsql.client", "roles/artifactregistry.reader", "roles/storage.admin"]
-  depends_on = [ module.iap_auth ]
+  roles               = concat(var.predefined_iam_roles, var.gcp_service_account_iam_roles)
+  depends_on          = [ module.iap_auth ]
+}
+
+resource "kubernetes_annotations" "hub" {
```

Comment:
this is not required, jupyterhub-workload-identity will create gcp and k8s sa with same name and configure annotations.